AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/resourcegroup/boundary/ResourceBoundary.java

```diff
@@ -287,4 +294,72 @@ public void updateResource(ResourceEntity resourceEntity) {
         }
     }
 
+    public void deleteApplicationServerById(int id) throws ResourceNotFoundException,
+            ResourceNotDeletableException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
+        doRemoveResourceEntity(ForeignableOwner.getSystemOwner(), id, false);
+    }
+
+    public void removeResource(ForeignableOwner deletingOwner, Integer resourceId) throws ResourceNotFoundException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
+        doRemoveResourceEntity(deletingOwner, resourceId, false);
+    }
+
+    public void removeResourceEntityOfDefaultResType(ForeignableOwner deletingOwner, Integer resourceId) throws ResourceNotFoundException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
+        doRemoveResourceEntity(deletingOwner, resourceId, true);
+    }
+
+    public void deleteApplicationById(ForeignableOwner deletingOwner, int applicationResId) throws ResourceNotFoundException,
+            ResourceNotDeletableException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
+        doRemoveResourceEntity(deletingOwner, applicationResId, false);
+    }
+
+    private void doRemoveResourceEntity(ForeignableOwner deletingOwner, Integer resourceId, boolean isDefaultResType) throws ResourceNotFoundException, ElementAlreadyExistsException, ForeignableOwnerViolationException {
+
+        ResourceEntity resourceEntity = commonService.getResourceEntityById(resourceId);
+
+        foreignableService.verifyDeletableByOwner(deletingOwner, resourceEntity);
+
+        if (resourceEntity == null) {
+            String message = "Die zu löschende Ressource ist nicht vorhanden";
```

Comment:
@yvespp Yes, without them as reference it would not be possible to display the original properties. (Like the name of a deleted environment or app server)